kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -98,6 +107,108 @@ public static Map<String, Map<String, Variant<?>>> buildSettings(NetworkProperti
         return newConnectionSettings;
     }
 
+    public static Map<String, Variant<?>> build8021xSettings(NetworkProperties props, String deviceId) {
+
+        String eap = props.get(String.class, "net.interface.%s.config.802-1x.eap", deviceId);
+        Optional<String> phase2 = props.getOpt(String.class, "net.interface.%s.config.802-1x.innerAuth", deviceId);
+
+        Map<String, Variant<?>> settings = new HashMap<>();
+
+        // Configure Eap Method
+        switch (Kura8021xEAP.valueOf(eap)) {
+        case ttls:
+            build8021xTunneledTls(props, deviceId, settings);
+            break;
+        case peap:
+            build8021xProtectedEap(props, deviceId, settings);
+            break;
+        case tls:
+            build8021xTls(props, deviceId, settings);
+            break;
+        default:
+            throw new IllegalArgumentException("Security type 802-1x \"" + eap + "\" is not supported.");
+        }
+
+        if (!phase2.isPresent()) {
+            return settings;
+        }
+
+        // Configure Phase2 (innerAuth) Method
+        switch (Kura8021xInnerAuth.valueOf(phase2.get())) {
+        case none:
+            break;
+        case mschapv2:
+            build8021xMschapV2(props, deviceId, settings);
+            break;
+        default:
+            throw new IllegalArgumentException("Security type 802-1x \"" + phase2 + "\" is not supported.");
+        }
+
+        return settings;
+    }
+
+    private static void build8021xTunneledTls(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.ttls.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xProtectedEap(NetworkProperties props, String deviceId,
+            Map<String, Variant<?>> settings) {
+        settings.put("eap", new Variant<>(new String[] { NM8021xEAP.peap.name() }));
+        build8021xOptionalCaCertAndAnonIdentity(props, deviceId, settings);
+    }
+
+    private static void build8021xTls(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
```

Comment:
Rename this method for consistency. We reserved `build*` for the top level public methods, `create*` for the private ones called by the `build*` ones.

```suggestion
    private static void create8021xTls(NetworkProperties props, String deviceId, Map<String, Variant<?>> settings) {
```